Neo4j.Driver/Neo4j.Driver.Tests.TestBackend/Protocol/Session/SessionRun.cs

```diff
@@ -25,32 +24,29 @@ public class SessionRunType
             [JsonProperty(Required = Required.AllowNull)]
             public Dictionary<string, object> txMeta { get; set; } = new Dictionary<string, object>();
 
-            [JsonProperty(Required = Required.AllowNull)]
-            public int timeout { get; set; } = -1;
-
+            public int? timeout { get; set; }
         }
 
         private Dictionary<string, object> ConvertParameters(Dictionary<string, CypherToNativeObject> source)
-		{
+        {
             if (data.parameters == null)
                 return null;
 
             Dictionary<string, object> newParams = new Dictionary<string, object>();
 
             foreach(KeyValuePair<string, CypherToNativeObject> element in source)
-			{
+            {
                 newParams.Add(element.Key, CypherToNative.Convert(element.Value));
-			}
+            }
 
             return newParams;
-		}
+        }
 
         void TransactionConfig(TransactionConfigBuilder configBuilder)
         {
-            if (data.timeout != -1)
+            if (data.timeout.HasValue)
             {
-                var time = TimeSpan.FromMilliseconds(data.timeout);
-                configBuilder.WithTimeout(time);
+                configBuilder.WithTimeout(TimeSpan.FromMilliseconds(data.timeout.Value));
             }
```

Comment:
I don't think this is doing what I envisioned it to do. There should be three cases covered here:
 * Set the timeout to `null` explicitly
 * Set the timeout to a given time span
 * Don't set the timeout at all

This code never runs into the first case.